frontend/packages/core-auth/src/components/AuthWrapper.tsx

```diff
@@ -114,16 +111,57 @@ const AuthWrapper: React.FC<AuthWrapperProps> = ({
     }
   }, [tokenResponse, isLoggedIn]);
 
+  // Schedule a token refresh before it expires (or use an existing schedule)
+  const refreshTokenTimeout = useRef<number | null>(null);
   useEffect(() => {
-    onTokenChange(token);
-  }, [token]);
+    if (
+      tokenResponse != null &&
```

Comment:
All of these should be `!==` unless there's a good reason

Reply:
Switch to truthy checks for object and `== / !=` for primitives. https://basarat.gitbook.io/typescript/styleguide#null-vs.-undefined